### PR TITLE
Add SqliteDefaultStoreSummaryHook

### PR DESCRIPTION
**Pull request by lann**
Also:

- Add `AppState: Sync` constraints to factors traits
- Stop `KeyValueDefaultStoreSummaryHook` printing for unused default store

### CODE REVIEW

File: crates/factor-sqlite/src/lib.rs [hunk 1]
```diff
@@ -205,4 +209,11 @@ pub trait Connection: Send + Sync {
     ) -> Result<v2::QueryResult, v2::Error>;
 
     async fn execute_batch(&self, statements: &str) -> anyhow::Result<()>;
+
+    /// A human-readable summary of the connection's configuration
+    ///
+    /// Example: "libSQL at libsql://example.com"
+    fn summary(&self) -> Option<String> {
```
Comment (rylev) [inline crates/factor-sqlite/src/lib.rs:216]: What's the thinking around returning `Option` here and defaulting this implementation? Seems like we're just asking for new implementors to forget to implement this method leading to subtle bugs. Why not force implementors to implement this?

**Review by rylev — APPROVED** [review]
Just one question, but we don't have to block merging this.

File: crates/factor-sqlite/src/lib.rs (hunk 1 above)
Comment (lann) [reply crates/factor-sqlite/src/lib.rs:216]: My thinking is that this is really only applicable to `spin up`, so if some other runtime has its own store impls it probably doesn't care about this method.

Comment (lann) [reply crates/factor-sqlite/src/lib.rs:216]: also the consequence for omitting this is small, at least as currently used

**Pull request merged**